cmake spdlog crash

hey im trying to recompile because of the recent CDLC and am getting a cmake failure, im on mint 20.1 and the .deb isnt working for whatever reason.
[CMakeError.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7572953/CMakeError.log)
[CMakeOutput.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7572954/CMakeOutput.log)

these are the files cmake says to look at, and on line 101 in CmakeError.log there's
``/home/stanbery/Games/Arma3/arma3-unix-launcher/build/CMakeFiles/CMakeTmp/src.cxx:1:10: fatal error: spdlog/spdlog.h: No such file or directory``

and help would be appreciated.

ok, starting with an empty folder i did
```
~/Games/Arma3$ git clone https://github.com/muttleyxd/arma3-unix-launcher.git
Cloning into 'arma3-unix-launcher'...
remote: Enumerating objects: 1790, done.
remote: Counting objects: 100% (322/322), done.
remote: Compressing objects: 100% (220/220), done.
remote: Total 1790 (delta 197), reused 176 (delta 100), pack-reused 1468
Receiving objects: 100% (1790/1790), 1.39 MiB | 4.93 MiB/s, done.
Resolving deltas: 100% (1140/1140), done.
~/Games/Arma3$ cd ./arma3-unix-launcher
~/Games/Arma3/arma3-unix-launcher$ mkdir build && cd build
~/Games/Arma3/arma3-unix-launcher/build$ cmake ..
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test SYSTEM_argparse_WORKS
-- Performing Test SYSTEM_argparse_WORKS - Failed
-- Downloading argparse from https://github.com/p-ranav/argparse.git
-- Using external argparse
-- Performing Test SYSTEM_fmt_WORKS
-- Performing Test SYSTEM_fmt_WORKS - Success
CMake Error at cmake/external_dependencies.cmake:92 (add_library):
  add_library cannot create ALIAS target "fmt::fmt-header-only" because
  target "fmt::fmt" is imported but not globally visible.
Call Stack (most recent call first):
  CMakeLists.txt:26 (setup_fmt)


-- Performing Test SYSTEM_pugixml_WORKS
-- Performing Test SYSTEM_pugixml_WORKS - Success
-- Performing Test SYSTEM_nlohmann_json_WORKS
-- Performing Test SYSTEM_nlohmann_json_WORKS - Failed
-- Performing Test SYSTEM_spdlog_WORKS
-- Performing Test SYSTEM_spdlog_WORKS - Success
-- Configuring incomplete, errors occurred!
See also "~/Games/Arma3/arma3-unix-launcher/build/CMakeFiles/CMakeOutput.log".
See also "~/Games/Arma3/arma3-unix-launcher/build/CMakeFiles/CMakeError.log".
```
these are the logs 
[CMakeError.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7576984/CMakeError.log)
[CMakeOutput.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7576985/CMakeOutput.log)

I just wanted to add that I'm also experiencing what seems to be the same problem with commit 309.

This is the output of cmake ..
```
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Performing Test SYSTEM_argparse_WORKS
-- Performing Test SYSTEM_argparse_WORKS - Failed
-- Downloading argparse from https://github.com/p-ranav/argparse.git
-- Using external argparse
-- Performing Test SYSTEM_fmt_WORKS
-- Performing Test SYSTEM_fmt_WORKS - Success
CMake Error at cmake/external_dependencies.cmake:92 (add_library):
  add_library cannot create ALIAS target "fmt::fmt-header-only" because
  target "fmt::fmt" is imported but not globally visible.
Call Stack (most recent call first):
  CMakeLists.txt:26 (setup_fmt)


-- Performing Test SYSTEM_pugixml_WORKS
-- Performing Test SYSTEM_pugixml_WORKS - Failed
-- Downloading pugixml from https://github.com/muttleyxd/pugixml.git
-- Using external pugixml
-- Performing Test SYSTEM_nlohmann_json_WORKS
-- Performing Test SYSTEM_nlohmann_json_WORKS - Failed
-- Performing Test SYSTEM_spdlog_WORKS
-- Performing Test SYSTEM_spdlog_WORKS - Success
-- Configuring incomplete, errors occurred!
See also "/home/[$redacted]/ThirdParty/arma3-unix-launcher/build/CMakeFiles/CMakeOutput.log".
See also "/home/[$redacted]/ThirdParty/arma3-unix-launcher/build/CMakeFiles/CMakeError.log".

```

All requirements to build from source are already installed.

```
cmake is already the newest version (3.16.3-1ubuntu1).
libfmt-dev is already the newest version (6.1.2+ds-2).
libqt5svg5 is already the newest version (5.12.8-0ubuntu1).
libqt5svg5-dev is already the newest version (5.12.8-0ubuntu1).
libqt5widgets5 is already the newest version (5.12.8+dfsg-0ubuntu1).
libspdlog-dev is already the newest version (1:1.5.0-1).
libspdlog1 is already the newest version (1:1.5.0-1).
qt5-default is already the newest version (5.12.8+dfsg-0ubuntu1).

```

[inxi.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7607383/inxi.log)
[CMakeError.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7607388/CMakeError.log)
[CMakeOutput.log](https://github.com/muttleyxd/arma3-unix-launcher/files/7607389/CMakeOutput.log)

--- a/cmake/external_dependencies.cmake
+++ b/cmake/external_dependencies.cmake
@@ -1,9 +1,10 @@
+
 include(CheckCXXSourceCompiles)
 include(FetchContent)
 
 function(setup_library SOURCE_TO_TEST)
     set(boolArgs HEADER_ONLY)
-    set(oneValueArgs NAME GIT_REPOSITORY GIT_TAG TEST_DEFINITIONS TEST_LINK_LIBS)
+    set(oneValueArgs NAME CXX_FLAGS GIT_REPOSITORY GIT_TAG TEST_DEFINITIONS TEST_LINK_LIBS)
     set(multiValueArgs WHEN)
     cmake_parse_arguments(LIB_SETUP "${boolArgs}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
@@ -36,6 +37,7 @@ function(setup_library SOURCE_TO_TEST)
         set(SRCDIR "${LIB_SETUP_NAME}_SOURCE_DIR")
         set(BINDIR "${LIB_SETUP_NAME}_BINARY_DIR")
 
+        string(APPEND CMAKE_CXX_FLAGS " ${LIB_SETUP_CXX_FLAGS}")
         add_subdirectory(${${SRCDIR}} ${${BINDIR}} EXCLUDE_FROM_ALL)
         message("-- Using external ${LIB_SETUP_NAME}")
     endif()
@@ -72,6 +74,9 @@ endfunction()
 function(setup_fmt)
     set(CHECK_SOURCE "#include <fmt/format.h>
         #include <string_view>
+
+        template<typename... Args>
+        using format_string_t = fmt::format_string<Args...>; // required by spdlog
 
         int main()
         {
@@ -152,6 +157,9 @@ function(setup_spdlog)
     set(SPDLOG_FMT_EXTERNAL OFF)
     set(SPDLOG_FMT_EXTERNAL_HO ON)
     set(CHECK_SOURCE "#include <spdlog/spdlog.h>
+        #if !__has_include(<spdlog/cfg/env.h>) // required by verboseness level setting
+        #error libspdlog 1.9 or newer is required
+        #endif
         int main()
         {
             spdlog::info(\"hello\");
@@ -169,6 +177,7 @@ function(setup_spdlog)
                   GIT_TAG v1.x
                   TEST_DEFINITIONS -DSPDLOG_FMT_EXTERNAL
                   TEST_LINK_LIBS ${FMT_TARGET_NAME}
+                  CXX_FLAGS "-Wno-attributes"
                   )
 endfunction()
 

